### PR TITLE
Gif Search in chat room

### PR DESCRIPTION
**Pull request by NickersWeb**
Left api key for giphy in the `libs/constants.js` ... change it to another api key when production. 
Left the ability to select a gif with an alert... not sure how you guys would want to handle this. Potentially a message type property that will illustrate it being a gif.

### CODE REVIEW

**Comment by amitojsingh366** [comment]
please follow the proper pattern for commit messages *(💡TIP: you can use `yarn commit` or `yarn commit:signed` and answer the questions asked to generate a properly formatted one for you!)*

to fix your most recent commit you can run the command `git commit --amend -m "properly formatted commit message here"`

to fix your older/multiple commits please take a look at [this article](https://docs.github.com/en/github/committing-changes-to-your-project/changing-a-commit-message#amending-older-or-multiple-commit-messages)

**Comment by amitojsingh366** [comment]
please don't use npm in kebab, all of our workspaces use yarn. also im not sure if we need a gif picker rn and if we do i don't think you did this the right way, you should probably create a new message token and pass it as that

**Review by willdoescode — CHANGES_REQUESTED** [review]
Rewrite in rust

**Comment by samarmohan** [comment]
> Rewrite in rust

lol but be serious please

**Comment by samarmohan** [comment]
Make sure all checks pass so we can merge. Your code is fine, style/guidelines are not.

**Comment by willdoescode** [comment]
> > Rewrite in rust
> 
> lol but be serious please

😎

**Comment by benawad** [comment]
idk if I want gifs in chat (I'm leaning towards no and being a twitch chat), but let me think about it

**Comment by willdoescode** [comment]
> idk if I want gifs in chat (I'm leaning towards no and being a twitch chat), but let me think about it

I agree with ben, you should also rewrite in rust

**Review by samarmohan — CHANGES_REQUESTED** [review]
Do we actually need this? @benawad

**Comment by ghost** [comment]
> 
> 
> > idk if I want gifs in chat (I'm leaning towards no and being a twitch chat), but let me think about it
> 
> I agree with ben, you should also rewrite in rust

Nah, we will rewrite it in C#.

**Comment by NickersWeb** [comment]
Happy to work towards the guidelines. Just a thought, perhaps this could be a room setting by the admin after they have created the room.

File: kebab/src/entities.ts [hunk 1]
```diff
@@ -14,7 +14,9 @@ export type UserPreview = {
   avatarUrl: string | null;
 };
 
+//enums
```
Comment (overlisted) [inline kebab/src/entities.ts:17]:
```suggestion
// enums
```

File: kebab/src/entities.ts [hunk 2]
```diff
@@ -131,3 +135,18 @@ export type CurrentRoom = Room & {
 };
 
 export type BooleanMap = Record<UUID, boolean>;
+
+export type Image = {
+  url: string;
+  width: number;
+  height: number;
+}
```
Comment (overlisted) [inline kebab/src/entities.ts:143]:
```suggestion
};

```

File: kebab/src/entities.ts [hunk 3]
```diff
@@ -131,3 +135,18 @@ export type CurrentRoom = Room & {
...
+export type GifResponse = {
+  id: string;
+  title: string;
+  images: {
+    original: Image;
+    original_still: Image;
+  }
+
+}
```
Comment (overlisted) [inline kebab/src/entities.ts:152]:
```suggestion
};
```

File: kebab/src/http/giphy.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+                data: GifResponse[]
+            }>,
+        queryGifs: (query: string, limit: number, rating: GifRating) =>
+            create({
+                baseUrl: 'https://api.giphy.com/v1/gifs/'
+            }).request('GET', `search?api_key=${apiKey}&q=${query}&limit=${limit}&rating=${rating}`) as Promise<{
+                data: GifResponse[]
+            }>
+    };
+};
```
Comment (overlisted) [inline kebab/src/http/giphy.ts:20]:
```suggestion
export const giphy = (apiKey: string) => ({
        trendingGifs: (limit: number, rating: GifRating) =>
            create({
                baseUrl: 'https://api.giphy.com/v1/gifs/'
            }).request('GET', `trending?api_key=${apiKey}&limit=${limit}&rating=${rating}`) as Promise<{
                data: GifResponse[]
            }>,
        queryGifs: (query: string, limit: number, rating: GifRating) =>
            create({
                baseUrl: 'https://api.giphy.com/v1/gifs/'
            }).request('GET', `search?api_key=${apiKey}&q=${query}&limit=${limit}&rating=${rating}`) as Promise<{
                data: GifResponse[]
            }>
});
```

Comment (overlisted) [inline kebab/src/http/giphy.ts:20]: what is this indentation

File: kebab/src/http/index.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
 export * from "./raw";
 export * from "./wrapper";
 export * as bot from "./bot";
+export * from './giphy';
```
Comment (overlisted) [inline kebab/src/http/index.ts:4]:
```suggestion
export * from "./giphy";
```

**Review by overlisted — CHANGES_REQUESTED** [review]
Use ESLint and Prettier please

File: kebab/package.json [hunk 6]
```diff
@@ -39,4 +39,4 @@
     "ts-jest": "^26.5.5",
     "typescript": "^4.2.4"
   }
-}
+}
```
Comment (overlisted) [inline kebab/package.json:42]:
```suggestion
}

```

File: kebab/src/entities.ts [hunk 7]
```diff
@@ -131,3 +135,19 @@ export type CurrentRoom = Room & {
...
+
+export type GifResponse = {
+  id: string;
+  title: string;
+  images: {
+    original: Image;
+    original_still: Image;
+  }
+
+}
```
Comment (overlisted) [inline kebab/src/entities.ts:153]:
```suggestion
}
```

File: kebab/src/http/giphy.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+                baseUrl: 'https://api.giphy.com/v1/gifs/'
+            }).request('GET', `trending?api_key=${apiKey}&limit=${limit}&rating=${rating}`) as Promise<{
+                data: GifResponse[]
+            }>,
+        queryGifs: (query: string, limit: number, rating: GifRating) =>
+            create({
+                baseUrl: 'https://api.giphy.com/v1/gifs/'
+            }).request('GET', `search?api_key=${apiKey}&q=${query}&limit=${limit}&rating=${rating}`) as Promise<{
+                data: GifResponse[]
+            }>
```
Comment (overlisted) [inline kebab/src/http/giphy.ts:16]:
```suggestion
  trendingGifs: (limit: number, rating: GifRating) =>
    create({
      baseUrl: 'https://api.giphy.com/v1/gifs/'
    }).request('GET', `trending?api_key=${apiKey}&limit=${limit}&rating=${rating}`) as Promise<{
      data: GifResponse[]
    }>,
  queryGifs: (query: string, limit: number, rating: GifRating) =>
    create({
      baseUrl: 'https://api.giphy.com/v1/gifs/'
    }).request('GET', `search?api_key=${apiKey}&q=${query}&limit=${limit}&rating=${rating}`) as Promise<{
      data: GifResponse[]
    }>
```

**Review by overlisted — APPROVED** [review]

**Comment by benawad** [comment]
decided I don't want this

**Pull request closed without merging**